File: FE/WEB/app/models/erp-site.js
import Model, { attr, belongsTo, hasMany } from '@ember-data/model';

export default class ErpSiteModel extends Model {
  @belongsTo('erp-site', { async: true, inverse: 'childSites' }) parentSite; // Relazione con la sede madre
  @hasMany('erp-site', { async: true, inverse: 'parentSite' }) childSites; // Relazione con le sotto-sedi

  @attr('number') tenantId; // Identificativo del tenant
  @attr('string', { defaultValue: '' }) name; // Nome della sede
  @attr('string') address; // Indirizzo della sede
  @attr('number') addressNumber; // Numero civico
  @attr('string') phone; // Numero di telefono
  @attr('string') city; // Città
  @attr('string') province; // Provincia
  @attr('string') zip; // Codice postale
  @attr('string') state; // Stato (es. 'IT', 'US')
  @attr('boolean', { defaultValue: false }) administrativeHeadquarters; // Sede amministrativa
  @attr('boolean', { defaultValue: false }) registeredOffice; // Sede legale
  @attr('boolean', { defaultValue: false }) operationalHeadquarters; // Sede operativa
  @attr('string') parentSiteId; // Guid della sede madre
  @attr('date-utc') createdAt; // Data di creazione del record
  @attr('date-utc', {
    defaultValue() {
      return new Date();
    },
  })
  updatedAt; // Data di ultima modifica del record
}

File: BE/Src/MIT.Fwk.Infrastructure/Entities/ErpSite.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace MIT.Fwk.Infrastructure.Entities
{
    [Table("ErpSite")]
    public class ErpSite : Identifiable<string>
    {
        [Attr] public int TenantId { get; set; }
        [Attr] public string Name { get; set; }
        [Attr] public string Address { get; set; }
        [Attr] public int? AddressNumber { get; set; }
        [Attr] public string Phone { get; set; }
        [Attr] public string City { get; set; }
        [Attr] public string Province { get; set; }
        [Attr] public string Zip { get; set; }
        [Attr] public string State { get; set; }
        [Attr] public bool AdministrativeHeadquarters { get; set; }
        [Attr] public bool RegisteredOffice { get; set; }
        [Attr] public bool OperationalHeadquarters { get; set; }
        [Attr] public string ParentSiteId { get; set; }
        [Attr] public DateTime CreatedAt { get; set; }
        [Attr] public DateTime UpdatedAt { get; set; }

        [HasOne] public virtual ErpSite ParentSite { get; set; }
        [HasOne] public virtual Tenant Tenant { get; set; }
    }

}
